### PR TITLE
EC errors from FS creation are fatal, and reconciliation should stop.

### PR DESCRIPTION
**Pull request by roehrich-hpe**
Signed-off-by: Dean Roehrich <dean.roehrich@hpe.com>

### CODE REVIEW

**Review by ajfloeder — APPROVED** [review]

**Review by bdevcich-hpe — APPROVED** [review]

File: controllers/nnf_node_storage_controller.go [hunk 1]
```diff
@@ -450,7 +450,7 @@ func (r *NnfNodeStorageReconciler) formatFileSystem(nodeStorage *nnfv1alpha1.Nnf
 		nodeStorage.Status.Error = dwsv1alpha1.NewResourceError("Could not create file system", err).WithFatal()
 		log.Info(nodeStorage.Status.Error.Error())
 
-		return &ctrl.Result{Requeue: true}, nil
+		return &ctrl.Result{}, nil
```
Comment (matthew-richerson) [inline controllers/nnf_node_storage_controller.go:453]: Can we do a RequeueAfter instead? This assumes that we can never recover.

Comment (roehrich-hpe) [reply controllers/nnf_node_storage_controller.go:453]: If it's an error in the args for zpool, mkfs, mount...then we can't recover.  The workflow has to be torn down, the storageprofile has to be fixed, and then a new workflow has to be created.

But you've put your comment on the place where the EC's data structures are being created.  Do we really expect that if the setup for the filesystem data structures didn't work the first time, it'll work the second time?

Comment (matthew-richerson) [reply controllers/nnf_node_storage_controller.go:453]: My comment was meant for both of the places you're taking out the requeue. There are errors other than invalid arguments that could be more transient. For example, a network issue preventing a Lustre OST/MDT mount from contacting the MGS.

Comment (matthew-richerson) [reply controllers/nnf_node_storage_controller.go:453]: Sounds good

**Review by matthew-richerson — APPROVED** [review]

**Pull request merged**